### PR TITLE
Ignore maxquota annotation if quota is disabled; refactor for testability

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
This is mainly a refactoring with a minor fix for an edge case: If quota is disabled, the annotation `service.dns.extensions.gardener.cloud/default-external-provider-entries-quota` is ignored now.

  Refactor: Extract and test DNS quota calculation logic

  Extracts the DNS default domain quota calculation from createOrUpdateDNSProviders into a standalone GetDefaultDomainQuota function with comprehensive test coverage.

  Changes:
  - Extracted quota calculation logic into testable function
  - Added 14 test cases using DescribeTable covering all scenarios (happy paths, edge cases, validation)
  - No functional changes - pure refactoring     
       
  Why: 
  - Improves code testability and maintainability
  - Provides clear unit tests for quota validation logic
 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
follow-up of #690 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/controller/lifecycle/actuator_quota_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
```
Comment (marc1404) [inline pkg/controller/lifecycle/actuator_quota_test.go]: nit: There are other files with the old timestamp in it, but I guess we don't need to add new ones with it :) 
```suggestion
// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
```

File: pkg/controller/lifecycle/actuator.go [hunk 2]
```diff
@@ -1182,3 +1167,34 @@ func oneOf(strs ...*string) string {
...
+			return 0, fmt.Errorf("failed to parse default external provider entries quota %s (shoot annotation %s): %w", annotatedValue, ShootDNSServiceDefaultExternalProviderEntriesQuotaAnnotation, err)
+		}
+		if parsedQuota < 1 {
+			return 0, fmt.Errorf("invalid default external provider entries quota %s (shoot annotation %s)", annotatedValue, ShootDNSServiceDefaultExternalProviderEntriesQuotaAnnotation)
+		}
+		maxQuota := cfg.DefaultExternalProviderEntriesQuotaMax
+		if maxQuota == 0 {
+			maxQuota = quota // restrict to default quota if no maximum quota is configured via flag, to avoid accidentally setting an unreasonably high quota via annotation
+		}
+		if quota > 0 && parsedQuota > int64(maxQuota) {
```
Comment (marc1404) [inline pkg/controller/lifecycle/actuator.go]: nit: Based on the previous `quota == 0` and `quota < 1` early-returns we could simplify this expression
```suggestion
		if parsedQuota > int64(maxQuota) {
```

File: pkg/controller/lifecycle/actuator.go [hunk 3]
```diff
@@ -1182,3 +1167,34 @@ func oneOf(strs ...*string) string {
 	}
 	return ""
 }
+
+// GetDefaultDomainQuota calculates the DNS entries quota for the default external provider.
+// It returns the configured default quota, which can be overridden via the shoot annotation
+// 'service.dns.extensions.gardener.cloud/default-external-provider-entries-quota'.
+// The annotation value is validated and constrained by DefaultExternalProviderEntriesQuotaMax.
+// Returns 0 if quotas are disabled (DefaultExternalProviderEntriesQuota == 0).
+func GetDefaultDomainQuota(cfg config.DNSServiceConfig, cluster *controller.Cluster) (int32, error) {
```
Comment (marc1404) [inline pkg/controller/lifecycle/actuator.go]: nit: Does this function need to be exported since it's only used within the `lifecycle` module?

**Review by marc1404 — COMMENTED** [review]
Thanks for improving the quota and max quota logic!
A few minor comments below

File: pkg/controller/lifecycle/actuator_quota_test.go (hunk 1 above)
Comment (MartinWeindel) [reply pkg/controller/lifecycle/actuator_quota_test.go]: fixed

File: pkg/controller/lifecycle/actuator.go (hunk 2 above)
Comment (MartinWeindel) [reply pkg/controller/lifecycle/actuator.go]: nice, thanks

File: pkg/controller/lifecycle/actuator.go (hunk 3 above)
Comment (MartinWeindel) [reply pkg/controller/lifecycle/actuator.go]: good idea

**Review by marc1404 — COMMENTED** [review]
Thanks for addressing my feedback!
/lgtm
/approve
